cmd/zdb/zdb.c

```diff
@@ -187,12 +187,22 @@ usage(void)
 	(void) fprintf(stderr, "        -I <number of inflight I/Os> -- "
 	    "specify the maximum number of "
 	    "checksumming I/Os [default is 200]\n");
+	(void) fprintf(stderr, "        -G dump zfs_dbgmsg buffer before "
+	    "exiting\n");
 	(void) fprintf(stderr, "Specify an option more than once (e.g. -bb) "
 	    "to make only that option verbose\n");
 	(void) fprintf(stderr, "Default is to dump everything non-verbosely\n");
 	exit(1);
 }
 
+static void
+dump_debug_buffer(void)
+{
+	if (dump_opt['G']) {
+		(void) printf("\n");
```

Comment:
This function was dropped in 3b36f8319d56842c85782e7842218a7499f3cf16 when the dbgmsg log was reworked for better integration with the Linux kernel.  Now that there's a user space consumer you can reintroduce it and it' prototype in the `include/sys/zfs_debug.h`.  Something like this:

```diff
diff --git a/module/zfs/zfs_debug.c b/module/zfs/zfs_debug.c
index b553d21..be28341 100644
--- a/module/zfs/zfs_debug.c
+++ b/module/zfs/zfs_debug.c
@@ -228,6 +228,21 @@ __dprintf(const char *file, const char *func, int line, con
 
        kmem_free(buf, size);
 }
+
+#else
+
+void
+zfs_dbgmsg_print(const char *tag)
+{
+       zfs_dbgmsg_t *zdm;
+
+       (void) printf("ZFS_DBGMSG(%s):\n", tag);
+       mutex_enter(&zfs_dbgmsgs_lock);
+       for (zdm = list_head(&zfs_dbgmsgs); zdm;
+               zdm = list_next(&zfs_dbgmsgs, zdm))
+       (void) printf("%s\n", zdm->zdm_msg);
+       mutex_exit(&zfs_dbgmsgs_lock);
+}
 #endif /* _KERNEL */
 
 #ifdef _KERNEL
```

You'll also want to manually verify that when passing `-G` to `zdb` the log does contain something.